### PR TITLE
[2783] Change default account name for ENS reversed-resolved name

### LINKED ISSUE #2783: Show ENS reverse-resolved name on account detail view
If a user has not changed their account nickname (it is not \Account $[0-9]+\), and an ENS reverse-resolved name is available for that account, then show the reverse-resolved name for that account.

Extension reference: https://github.com/MetaMask/metamask-extension/issues/11273

Requirements:
- Only show ENS reverse-resolved name in the account name (not in the address).

**Account Overview**
AccountOverview

**Accounts List**

### PR DESCRIPTION
**Pull request by gantunesr**
## Description

If a user has not changed their account nickname (it is not \Account $[0-9]+), and an ENS reverse-resolved name is available for that account, then show the reverse-resolved name.

## To Do

- [x] Default-ENS change in account overview
- [x] Default-ENS change in accounts list
- [x] Remove ENS from account overview ethereum address
- [x] Update testing

### CODE REVIEW

File: app/components/UI/AccountList/AccountElement/index.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ class AccountElement extends PureComponent {
 
 	render() {
 		const { disabled, updatedBalanceFromStore } = this.props;
```
Comment (wachunei) [inline app/components/UI/AccountList/AccountElement/index.js:121]: `ticker` is destructured again on line 123, I suggest adding it on this line

```suggestion
		const { disabled, updatedBalanceFromStore, ticker } = this.props;
```

File: app/components/UI/AccountList/AccountElement/index.js [hunk 2]
```diff
@@ -119,7 +119,7 @@ class AccountElement extends PureComponent {
...
-		const { address, name, isSelected, isImported, balanceError } = this.props.item;
+		const { address, name, ens, isSelected, isImported, balanceError } = this.props.item;
 		const { ticker } = this.props;
```
Comment (wachunei) [inline app/components/UI/AccountList/AccountElement/index.js:123]:
```suggestion
```

File: app/components/UI/AccountList/index.js [hunk 3]
```diff
@@ -319,6 +327,23 @@ class AccountList extends PureComponent {
...
+		const { network } = this.props;
+		orderedAccounts.forEach(async account => {
+			try {
+				const ens = await doENSReverseLookup(account.address, network);
+				this.setState({
+					accountsENS: {
+						...this.state.accountsENS,
+						[account.address]: ens
+					}
+				});
```
Comment (wachunei) [inline app/components/UI/AccountList/index.js:340]: Fix: `setState` takes a function with `state, props` as arguments ([reference](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous))
```suggestion
				this.setState(state => {
					accountsENS: {
						...state.accountsENS,
						[account.address]: ens
					}
				});
```

File: app/components/UI/AccountList/AccountElement/index.js [hunk 4]
```diff
@@ -129,6 +129,7 @@ class AccountElement extends PureComponent {
 				</Text>
 			</View>
 		) : null;
+		const regExp = /Account \d*$/;
```
Comment (rickycodes) [inline app/components/UI/AccountList/AccountElement/index.js:132]: i notice we have this same regExp in two spots, maybe we should add another utility method something like `isAccountName` or similar to the existing ens utils and then that way this regEx and logic can live in one place and be easily tested from there as well.

**Review by rickycodes — CHANGES_REQUESTED** [review]
one small suggestion re: DRY (don't repeat yourself)

File: app/components/UI/AccountList/AccountElement/index.js (hunk 1 above)
Comment (gantunesr) [reply app/components/UI/AccountList/AccountElement/index.js]: Done 👍

File: app/components/UI/AccountList/AccountElement/index.js (hunk 2 above)
Comment (gantunesr) [reply app/components/UI/AccountList/AccountElement/index.js]: Done 👍

File: app/components/UI/AccountList/index.js (hunk 3 above)
Comment (gantunesr) [reply app/components/UI/AccountList/index.js]: Fixed!

File: app/components/UI/AccountList/AccountElement/index.js (hunk 4 above)
Comment (gantunesr) [reply app/components/UI/AccountList/AccountElement/index.js]: Fixed! It's only pending the test to check the logic

**Comment by rickycodes** [comment]
looks like a number of snapshot tests are now failing. you can _usually_ update these via `yarn test:unit:update`

**Review by rickycodes — APPROVED** [review]
code LGTM now
thanks for adding the tests!

**Review by cortisiko — CHANGES_REQUESTED** [review]
Found an issue:

I notice when you tap on the burger menu the ENS name is missing. 
see [iOS](https://www.screencast.com/t/PxM4lpR0S0 ) and [android](https://www.screencast.com/t/aMX4mbk5ze) 

to reproduce:
log in using your ENS wallet
tap on the burger menu
notice the wallet address appears instead of your ENS name.

**Review by cortisiko — APPROVED** [review]
🚀

**Pull request merged**